src/algebra/big_operators/finprod.lean

```diff
@@ -0,0 +1,623 @@
+/-
+Copyright (c) 2020 Kexing Ying and Kevin Buzzard. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying, Kevin Buzzard, Yury Kudryashov
+-/
+
+import data.set.finite
+import data.set.disjointed
+import algebra.big_operators
+import data.indicator_function
+
+/-!
+# Finite products and sums over types and sets
+
+We define products sums over types and subsets of types, with no finiteness hypotheses. All infinite
+product and sums are defined to be junk values (i.e. one or zero). This approach is sometimes easier
+to use than `finset.sum` when issues arise with `finset` and `fintype` being data.
+
+## Main definitions
+
+We use the following variables:
+
+* `α`, `β` - types with no structure;
+* `s`, `t` - sets
+* `M`, `N` - additive or multiplicative commutative monoids
+* `f`, `g` - functions
+
+Definitions in this file:
+
+* `finsum f : M` : the sum of `f x` as `x` ranges over the support of `f`, if it's finite.
+   Zero otherwise.
+
+* `finsum_in s f : M` : the sum of `f x` as `x` ranges over the elements of `s` for which `f x ≠ 0`,
+  if this is a finite sum. Zero otherwise.
+
+* `finprod f : M` : the product of `f x` as `x` ranges over the multiplicative support of `f`, if
+   it's finite. One otherwise.
+
+* `finprod_in s f : M` : the product of `f x` as `x` ranges over the elements of `s` for which
+  `f x ≠ 1`, if this is a finite sum. One otherwise.
```

Comment:
These no longer exist, right?
```suggestion
* `finsum f : M` : the sum of `f x` as `x` ranges over the support of `f`, if it's finite.
   Zero otherwise.

* `finprod f : M` : the product of `f x` as `x` ranges over the multiplicative support of `f`, if
   it's finite. One otherwise.
```